tools/check-upcoming-maintenance.py

```diff
@@ -0,0 +1,130 @@
+import os
+import re
+import subprocess
+from typing import List
+from optparse import OptionParser
+
+class NodeMaintenance:
+    '''
+    Class to keep track of project, zones, and search terms for nodes, as well as results from gcloud queries
+    '''
+    def __init__(self, project, zones, regex):
+        self.project = project
+        res = subprocess.run("gcloud projects describe {}".format(self.project), shell=True, capture_output=True, text=True)
```

Comment:
There are multiple side-effects in this constructor. It's better to have `__init__` without side effects.
I would adivce to have a factory-method that takes you current `__init__` signature and build an instance, while 
change `__init__` contract to take already fetched&parsed state

Reply:
Done. Let me know if you had something else in mind.